recipe/conda_build_config.yaml

```diff
@@ -47,6 +60,25 @@ VERBOSE_AT:
   - V=1
 VERBOSE_CM:
   - VERBOSE=1
+
+# dual build configuration
+channel_sources:
+  - conda-forge,defaults
+  - conda-forge/label/new,defaults  # [linux or osx]
+
+channel_targets:
+  - conda-forge main
+  - conda-forge new                             # [linux or osx]
+
+docker_image:                                   # [linux or ppc64le]
+  - condaforge/linux-anvil                      # [linux]
+  - conda/c3i-linux64                           # [linux]
+  - condaforge/ppc-anvil                        # [ppc64le]
+
+build_number_increment:                         # [linux or osx]
+  - 0                                           # [linux or osx]
+  - 1000                                        # [linux or osx]
```

Comment:
Please help me understand this. IIUC, this is so we avoid build number collisions if `main` and `new` were to be merged and to express a preference (due to higher numbers) for the `new` builds. But how would that be implemented? Let's work on an example:
Say there is an existing recipe and build for `pkg` such that we have the following `meta.yaml`
```yaml
package:
  name: pkg
  version: 0.1
build:
  number: 0
```
and a build
```
conda-forge/label/main::pkg=0.1=0
```
. IIUC, you want builds for `new` to implicitly add `1000` to the build number such that we'll get
```
conda-forge/label/new::pkg=0.1=1000
```
and for subsequent builds with recipe
```yaml
package:
  name: pkg
  version: 0.1
build:
  number: 1
```
we get
```
conda-forge/label/main::pkg=0.1=1
conda-forge/label/new::pkg=0.1=1001
```
. But if `1000` is *added* implicitly, that might bite you when you decide to flip the switch, merge the channels/labels, remove that `build_number_increment` key, and then build
```yaml
package:
  name: pkg
  version: 0.1
build:
  number: 2
```
, which gets us
```
conda-forge/label/new::pkg=0.1=2
```
for which `build_number=2` is clearly lower that `build_number=1001` of the previous build.
So, if you want to use such a workaround that implicitly changes the build number, I'd rather have it
```yaml
build_number_decrement:
  - 1000
  - 0
```
and and change the existing recipe to
```yaml
package:
  name: pkg
  version: 0.1
build:
  number: 1000
```
(for the case when `build/number` is still `0`).
That's not pretty either, but it at least avoids the problem with `2 < 1001` in my example.
(I did a quick test to see what happens when you would use a decrement but forget to increase the build number in the recipe beforehand. And it looks like `conda-build` does indeed fail to build if the number is `< 0`, which is good and what I would expect it to do -- but sadly it doesn't fail gracefully..)

----
An (I assume, better) alternative is, of course, to use `build_number_increment` as proposed in this PR, but then add some logic in the build (and linting) process which checks whether a build with a higher number already exists in the channel and if so fail gracefully.

Reply:
:+1: to this suggestion. One more suggestion. If actual build number is < 1000 in the recipe, use only the older compilers. (This can be done by filtering the configs.)